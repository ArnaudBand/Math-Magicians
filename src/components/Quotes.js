import React from 'react';
import './css/quotes.css';

const Quotes = () => {
  const [quotes, setQuotes] = React.useState([]);
  const [randomQuote, setRandomQuote] = React.useState('');

  React.useEffect(() => {
    async function fetchData() {
      const response = await fetch('https://type.fit/api/quotes');
      const data = await response.json();

      setQuotes(data);
      const randIndex = Math.floor(Math.random() * data.length);
      setRandomQuote(data[randIndex]);
    }
    fetchData();
  }, []);

  const getNewQuote = () => {
    const randIndex = Math.floor(Math.random() * quotes.length);
    setRandomQuote(quotes[randIndex]);
  };

  return (
    <div style={{ backgroundColor: 'transparent', minHeight: '100vh' }}>
      <div className="jumbotron">
        <div className="card">
          <div className="card-body">
            {randomQuote ? (
              <>
                <p className="card-text">
                  &quot;
                  {randomQuote.text}
                  &quot;
                </p>
                <span className="card-title">
                  - Author:
                  {' '}
                  {randomQuote.author || 'No Author'}
                </span>
              </>
            ) : (
              <h2>Loading</h2>
            )}
          </div>
          <div className="div_btn">
            <button type="button" onClick={getNewQuote} className="btn_quotes">New Quote</button>
          </div>
        </div>

      </div>
      <p className="text-center">By Arnaud</p>
    </div>

  );
};

export default Quotes;
