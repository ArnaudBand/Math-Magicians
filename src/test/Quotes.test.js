import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

it('Quotes renders correctly', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
