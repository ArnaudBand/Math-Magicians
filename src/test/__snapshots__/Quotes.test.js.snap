// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "transparent",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="jumbotron"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <h2>
          Loading
        </h2>
      </div>
      <div
        className="div_btn"
      >
        <button
          className="btn_quotes"
          onClick={[Function]}
          type="button"
        >
          New Quote
        </button>
      </div>
    </div>
  </div>
  <p
    className="text-center"
  >
    By Arnaud
  </p>
</div>
`;
